tensorflow/python/eager/backprop.py

```diff
@@ -209,8 +209,12 @@ def grad_fn(*args, **kwds):
     if not variables:
       raise ValueError("No trainable variables were accessed while the "
                        "function was being computed.")
-
+    for v in variables:  
+      if is_floating(v) == False:
```

Comment:
"if not is_floating(v)" or the linter will fail

Reply:
This check probably belongs in `GradientTape.watch` before we even get to backprop. Wdyt?